extendedcopy.go

```diff
@@ -194,7 +194,25 @@ func (opts *ExtendedCopyGraphOptions) FilterAnnotation(key string, regex *regexp
 		}
 		var filtered []ocispec.Descriptor
 		for _, p := range predecessors {
-			if regex.MatchString(p.Annotations[key]) {
+			if p.Annotations == nil || p.Annotations[key] == "" {
```

Comment:
In Golang, this can be simplified as:
```suggestion
			if p.Annotations[key] == "" {
```

Or consider,
` if val, ok := p.Annotations[key]; ok {}`
See https://go.dev/blog/maps.

But what do we want to do for the 1.b scenario in below?
1. The descriptor has annotations
    a. The annotation map contains the desired key
     - Check if the value matches the regex
   
    b. The annotation map does not contain the desired key
     - Do we still need to parse the content in this case?
2. The descriptor has no annotation
     - Parse the annotations from the content

Reply:
Good question. The current implementation is to parse the content, since `if p.Annotations[key] == ""` represents both cases of no annotation and annotation not having the desired key. I think this is fine? @shizhMSFT

Reply:
Used the other logic posted in the group, so this issue does not exist anymore.